src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp

```diff
@@ -768,72 +763,50 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         }
     }
 
-    // Method Description:
-    // - Check if the bell sounds exist on disk. Mark any that don't exist
-    //   so that they show the appropriate UI
-    safe_void_coroutine ProfileViewModel::_CheckBellSoundsExistence()
+    BellSoundViewModel::BellSoundViewModel(const Model::IMediaResource& resource) :
+        _resource{ resource }
     {
-        co_await winrt::resume_background();
-        std::vector<Editor::BellSoundViewModel> markedSounds;
-        for (auto&& sound : _CurrentBellSounds)
-        {
-            if (!std::filesystem::exists(std::wstring_view{ sound.Path() }))
-            {
-                markedSounds.push_back(sound);
-            }
-        }
-
-        co_await winrt::resume_foreground(_dispatcher);
-        for (auto&& sound : markedSounds)
+        if (_resource.Ok() && _resource.Path() != _resource.Resolved())
         {
-            get_self<BellSoundViewModel>(sound)->FileExists(false);
+            // If the resource was resolved to something other than its path, show the path!
+            _ShowDirectory = true;

```

Comment:
@carlos-zamora since you wrote bell sounds, you might care about this. it means that if you use a relative path to your bell sound file, the settings UI will always display the folder it _resolved_ to